File: backend/app/schemas/output.py
"""Pydantic schemas for ReviewOutput — request/response models."""

from datetime import datetime

from pydantic import BaseModel

from app.models.enums import CitationStyle, OutputType


class ReviewOutputResponse(BaseModel):
    """Review output detail response."""

    model_config = {"from_attributes": True}

    id: str
    project_id: str
    output_type: OutputType
    title: str | None
    outline: dict | None
    content: str | None
    structured_data: dict | None
    references: list[dict] | None
    version: int
    language: str
    citation_style: CitationStyle
    writing_style: str | None
    citation_verification: list[dict] | None
    export_formats: list[str] | None
    created_at: datetime
    updated_at: datetime


class PaginatedResponse(BaseModel):
    """Unified paginated response wrapper."""

    items: list
    total: int
    page: int
    size: int
    pages: int

File: frontend/src/types/output.ts
import type { OutputType, CitationStyle } from './enums';

/** 引用验证条目 */
export interface CitationVerification {
    reference_index: number;
    status: 'verified' | 'unverified';
    title: string;
    doi: string | null;
}

/** 综述输出详情 — 对齐后端 ReviewOutputResponse */
export interface ReviewOutputResponse {
    id: string;
    project_id: string;
    output_type: OutputType;
    title: string | null;
    outline: Record<string, unknown> | null;
    content: string | null;
    structured_data: Record<string, unknown> | null;
    references: Record<string, unknown>[] | null;
    version: number;
    language: string;
    citation_style: CitationStyle;
    writing_style: string | null;
    citation_verification: CitationVerification[] | null;
    export_formats: string[] | null;
    created_at: string;
    updated_at: string;
}
